src/jllvm/vm/JIT.hpp

```diff
@@ -140,70 +71,37 @@ class JIT
     /// Add callable 'f' as implementation for symbol 'symbol' to the implementation library.
     template <class F>
     void addImplementationSymbol(std::string symbol, const F& f)
-        requires(!std::is_pointer_v<F> && !std::is_function_v<F>)
     {
-        llvm::cantFail(m_implDetails.define(
-            createLambdaMaterializationUnit(std::move(symbol), m_optimizeLayer, f, m_dataLayout, m_interner)));
+        llvm::cantFail(m_javaJITImplDetails.define(createLambdaMaterializationUnit(
+            std::move(symbol), m_byteCodeCompileLayer.getBaseLayer(), f, m_byteCodeCompileLayer.getDataLayout(),
+            m_byteCodeCompileLayer.getInterner())));
     }
 
-    /// Add function pointer 'f' as implementation for symbol 'symbol' to the implementation library.
-    template <class Ret, class... Args>
-    void addImplementationSymbol(llvm::StringRef symbol, Ret (*f)(Args...))
-    {
-        llvm::cantFail(m_implDetails.define(llvm::orc::absoluteSymbols(
-            {{m_interner(symbol), llvm::JITEvaluatedSymbol::fromPointer(f, llvm::JITSymbolFlags::Exported
-                                                                               | llvm::JITSymbolFlags::Callable)}})));
-    }
+    static std::unique_ptr<std::uint64_t[]> createOSRBuffer(llvm::ArrayRef<std::uint64_t> locals,
+                                                            llvm::ArrayRef<std::uint64_t> operandStack);
 
-    /// Add 'ptr' as implementation of global 'symbol' to the implementation library.
-    template <class T>
-    void addImplementationSymbol(llvm::StringRef symbol, T* ptr) requires(!std::is_function_v<T>)
-    {
-        llvm::cantFail(m_implDetails.define(
-            llvm::orc::absoluteSymbols({{m_interner(symbol), llvm::JITEvaluatedSymbol::fromPointer(ptr)}})));
-    }
+public:
+    explicit JIT(VirtualMachine& virtualMachine);
 
     /// Adds and registers a class in the JIT. This has to be done prior to being able to lookup and execute
```

Comment:
Comments should be updated.

Reply:
I chose to remove the comment since its now an interface implementation with a proper comment in the super class